influxql/parser.go

```diff
@@ -2057,7 +2058,7 @@ func (p *Parser) peekRune() rune {
 	return r
 }
 
-func (p *Parser) parseSource() (Source, error) {
+func (p *Parser) parseSource(subqueries bool) (Source, error) {
```

Comment:
Feels like there should be two methods here `parseSource` and `parseSourceWithSubqueries`.